routing/result_interpretation.go

```diff
@@ -517,11 +517,15 @@ func (i *interpretedResult) processPaymentOutcomeIntermediate(route *mcRoute,
 		if introIdx == len(route.hops.Val)-1 {
```

Comment:
not penalising here tho could have a side-effect, if we have a blinded path with 2 hops and the final hop fails, that we currently try forever until timeout in our payment flow.

Reply:
maybe we need to be more strict in our payment logic, to stop retrying way sooner.